internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -827,13 +837,43 @@ func NewKeystoreSecretForCR(cr *operatorv1beta1.Cryostat) *corev1.Secret {
 	}
 }
 
-func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat) *corev1.ServiceAccount {
+func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat, isOpenShift bool) (*corev1.ServiceAccount, error) {
+	if isOpenShift {
+		OAuthRedirectReference := &oauthv1.OAuthRedirectReference{
+			Reference: oauthv1.RedirectReference{
+				Kind: "Route",
+				Name: cr.Name,
+			},
+		}
+
+		ref, err := json.Marshal(OAuthRedirectReference)
+		if err != nil {
+			return nil, err
+		}
+
+		sa := &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cr.Name,
+				Namespace: cr.Namespace,
+				Labels: map[string]string{
+					"app": "cryostat",
+				},
+				Annotations: map[string]string{
+					"serviceaccounts.openshift.io/oauth-redirecturi.route":       "https://",
+					"serviceaccounts.openshift.io/oauth-redirectreference.route": string(ref),
+				},
+			},
+		}
+		return sa, nil
+	}
```

Comment:
Just one more suggestion to reduce duplication, you could:
1. Create an empty annotations map
2. Put the OAuth annotations in the map inside the OpenShift branch
3. Return the ServiceAccount with annotations set to the map. In Kubernetes this will be empty, in OpenShift it'll contain the OAuth annotations